File: SchoolManagement/SchoolManagement.Domain/Student.cs
using Sekiban.Pure.Aggregates;

namespace SchoolManagement.Domain;

[GenerateSerializer]
public record Student(
    string Name,
    string StudentId,
    DateTime DateOfBirth,
    string Email,
    string PhoneNumber,
    string Address,
    Guid? ClassId = null
) : IAggregatePayload
{
    public int GetAge()
    {
        var today = DateTime.Today;
        var age = today.Year - DateOfBirth.Year;
        if (DateOfBirth.Date > today.AddYears(-age)) age--;
        return age;
    }
}

[GenerateSerializer]
public record DeletedStudent(
    string Name,
    string StudentId,
    DateTime DateOfBirth,
    string Email,
    string PhoneNumber,
    string Address,
    Guid? ClassId = null
) : IAggregatePayload;

File: SchoolManagement/school-management-next/src/types/student.ts
export interface Student {
  studentId: string;
  name: string;
  studentIdNumber: string;
  dateOfBirth: string;
  email: string;
  phoneNumber: string;
  address: string;
  classId: string | null;
  age: number;
}

export interface RegisterStudentPayload {
  name: string;
  studentId: string;
  dateOfBirth: string;
  email: string;
  phoneNumber: string;
  address: string;
}

export interface UpdateStudentPayload {
  studentId: string;
  name?: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
}

export interface AssignStudentToClassPayload {
  studentId: string;
  classId: string;
}

export interface RemoveStudentFromClassPayload {
  studentId: string;
}
